mitiq/lre/tests/test_multivariate_richardson.py

```diff
@@ -0,0 +1,239 @@
+    assert (expected_matrix == sample_matrix(test_circ, test_degree, 1)).all()
+
+
+@pytest.mark.parametrize(
+    "test_circ, test_degree, test_fold_multiplier, expected_matrix",
+    [
+        (
+            test_circuit1,
+            2,
+            3,
+            [
+                1.5555555555555578,
+                -0.38888888888888934,
+                -0.38888888888888934,
+                0.09722222222222215,
+                0.027777777777777804,
+                0.09722222222222232,
+            ],
+        ),
+        (
+            test_circuit2,
+            2,
+            2,
+            [
+                2.4062499999999956,
+                -0.6874999999999987,
+                -0.6874999999999987,
+                -0.6874999999999987,
+                0.15624999999999956,
+                0.06249999999999997,
+                0.06249999999999997,
+                0.15624999999999956,
+                0.06249999999999997,
+                0.15624999999999956,
+            ],
+        ),
+    ],
+)
+def test_coeffs(test_circ, test_degree, test_fold_multiplier, expected_matrix):
+    assert expected_matrix == linear_combination_coefficients(
+        test_circ, test_degree, test_fold_multiplier
+    )
+
+
+@pytest.mark.parametrize(
+    "test_input, test_degree, test_fold_multiplier, error_msg",
+    [
+        (
+            test_circuit1,
+            0,
+            1,
+            "Multinomial degree must be greater than or equal to 1.",
+        ),
+        (
+            test_circuit1,
+            1,
+            0,
+            "Fold multiplier must be greater than or equal to 1.",
+        ),
+    ],
+)
+def test_invalid_degree_fold_multiplier_sample_matrix(
+    test_input, test_degree, test_fold_multiplier, error_msg
+):
+    """Ensures that the args for the sample matrix
+    an error for an invalid value."""
+    with pytest.raises(ValueError, match=error_msg):
+        sample_matrix(test_input, test_degree, test_fold_multiplier)
+
+
+@pytest.mark.parametrize(
+    "test_input, degree, test_fold_multiplier",
+    [
+        (test_circuit1, 1, 1),
+        (test_circuit1, 2, 1),
+        (test_circuit1, 3, 5),
+        (test_circuit1, 4, 7),
+        (test_circuit1, 2, 2),
+        (test_circuit1, 2, 3),
+    ],
+)
+def test_square_sample_matrix(test_input, degree, test_fold_multiplier):
+    """Check if the sample matrix will always be a square.
+
+    The terms in the monomial basis define the total rows of the sample matrix
+    & the generated scale factors for some fold multiplier define the number of
+    columns.
+    """
+    num_layers = _get_num_layers_without_measurements(test_input)
+    calculated_basis = full_monomial_basis(num_layers, degree)
+    calculated_scale_factor_vectors = _get_scale_factor_vectors(
+        test_input, degree, test_fold_multiplier
+    )
+    assert len(calculated_basis) == len(calculated_scale_factor_vectors)
+
+
+def test_lre_inference_with_chunking():
```

Comment:
We could probably also include tests that make use of the `num_chunks` parameter:

```py
@pytest.mark.parametrize("num_chunks", [None, 2, 4])
def test_linear_combination_coefficients_with_chunking(num_chunks):
    coeffs = linear_combination_coefficients(test_circuit2, 2, 2, num_chunks)
    assert len(coeffs) == _get_num_layers_without_measurements(test_circuit2)
    assert np.isclose(sum(coeffs), 1.0)  # Coefficients should sum to 1
```

Reply:
I'll try a different method of implementing this. 

`_get_num_layers_without_measurements` does not use `num_chunks`.

https://github.com/unitaryfund/mitiq/blob/231319df9051dfbc61db79d02bee201979528a36/mitiq/lre/multivariate_scaling/layerwise_folding.py#L23